cmd/run.go

```diff
@@ -211,7 +217,7 @@ func execute(cmd *cobra.Command, args []string) {
 		rerun.WritePrevArgs(os.Args)
 	}
 	installMissingPlugins(installPlugins)
-	exitCode := execution.ExecuteSpecs(specs)
+	exitCode = execution.ExecuteSpecs(specs, maxRetriesCount)
```

Comment:
Instead of passing the maxRetriesCount around, you can add set it in the execution package before. This is the approach we have used here. 
This https://github.com/getgauge/gauge/blob/master/cmd/cmd.go#L177 method inits all package flags before execution.